serve/mlc_serve/model/paged_cache_model.py

```diff
@@ -560,6 +625,7 @@ def generate(
                                     ),
                                     generated_tokens=[],
                                     error=err_msg,
+                                    logprob_info=logprob_infos[i]
```

Comment:
L606, 615, and 628 should be

```
logprob_info=logprob_infos[0]
```

This code path samples a token sequentially, so `logprob_infos.shape[0]` is always 1.

Reply:
It is correct. In this case index `i` means the number of sequence_id corresponded to request from list of requests (in batch case). If we use 0 instead we send responses with logprobs from token for the first request

Reply:
You need to study this code path more carefully. `len(logprob_infos)` is always 1, so `logprob_infos[i]` would be an out of bound access. See also https://github.com/octoml/mlc-llm/blob/46f0de68f69314e2f15f44a3332cdcfd3eea8580/serve/mlc_serve/model/paged_cache_model.py#L593

Reply:
Oh, yeah! I see `sample()` function with `logits_per_token` inside the loop. Thank you for remark!

Reply:
Hello @zxybazh! Could you add the following diff too:
```
diff --git a/serve/mlc_serve/model/paged_cache_model.py b/serve/mlc_serve/model/paged_cache_model.py
index c663dec..758afeb 100644
--- a/serve/mlc_serve/model/paged_cache_model.py
+++ b/serve/mlc_serve/model/paged_cache_model.py
@@ -603,7 +603,7 @@ class Model:
                                     ),
                                     generated_tokens=[new_token],  # type: ignore
                                     error=None,
-                                    logprob_info=logprob_infos[i]
+                                    logprob_info=logprob_infos[0]
                                 )
                             )
                     else:
@@ -612,7 +612,7 @@ class Model:
                                 sequence_id=sequence_id,
                                 generated_tokens=[new_token],  # type: ignore
                                 error=None,
-                                logprob_info=logprob_infos[i]
+                                logprob_info=logprob_infos[0]
                             )
                         )
                 else:
@@ -625,7 +625,7 @@ class Model:
                                     ),
                                     generated_tokens=[],
                                     error=err_msg,
-                                    logprob_info=logprob_infos[i]
+                                    logprob_info=logprob_infos[0]
                                 )
                             )
                     else:
@@ -634,7 +634,7 @@ class Model:
                                 sequence_id=sequence_id,
                                 generated_tokens=[],
                                 error=err_msg,
-                                logprob_info=logprob_infos[i]
+                                logprob_info=logprob_infos[0]
                             )
                         )
 
```